ReportPortal.ATM/ApiClientLibrary/Factories/RestClientServiceFactory.cs

```diff
@@ -0,0 +1,27 @@
+﻿using ApiClientLibrary.ApiClients;
+using ApiClientLibrary.Configurations;
+using ConfigurationLibrary.Interfaces.Configuration;
+using RestClientLibrary.Enums;
+using RestClientLibrary.Interfaces.Clients;
+using RestClientLibrary.Interfaces.Factories;
+
+namespace RestClientLibrary.Factories;
+
+public class RestClientServiceFactory : IRestClientServiceFactory
+{
+    public Interfaces.Clients.IRestClientService Create(RestClientServiceType restClientServiceType, IConfigurationService configurationService)
+    {
+        var apiClientConfiguration = new ApiClientConfiguration(configurationService);
```

Comment:
We should not be creating an instance of the `ApiClientConfiguration`. We have a few options here:

1. If the `IConfigurationService` is a singleton, then you can have a constructor method in this class and receive directly an `IApiClientConfiguration`.
    1. The Dependency Injection system will then use the `IConfigurationService` it already knows, to instanciate the `IApiClientConfiguration` for you.
2. If the `IConfigurationService` is not a singleton, then:
    1. Create an `IApiClientConfigurationFactory`.
    2. Receive an `IConfigurationServiceFactory` in its constructor.
    3. Implement an `IApiClientConfiguration Create (string fileName, string filePath)` method that will use the `IConfigurationServiceFactory` to create a new `IConfigurationService`.
    4. Use that `IConfigurationService` to create the `ApiClientConfiguration`.
    5. Create a constructor method for the `RestClientServiceFactory` that receives an `IApiClientConfigurationFactory`.
    6. Change the `Create` method's parameters to receive a `string filePath, string fileName, RestClientServiceType restClientServiceType`.

That way, the Dependency Injection system is going to handle all those dependencies and you will only deal on interfaces and abstractions.